docs/core/extensions/console-log-formatter.md

```diff
@@ -0,0 +1,429 @@
+
+```
+dotnet new webapp -o SampleWebApp
+```
+
+When running the app, using the template code, we get the default log format below:
+
+```
+info: Microsoft.Hosting.Lifetime[0]
+      Now listening on: https://localhost:5001
+```
+
+By default, the Simple console log formatter is selected with default configuration. To change from default, in Program.cs as seen below we could configure logging by adding `AddJsonConsole` which allows to select and configure the Json console formatter:
+
+```c#
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using System.Text.Json;
+using Microsoft.Extensions.WebEncoders.Testing;
+
+namespace SampleWebApp
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            CreateHostBuilder(args).Build().Run();
+        }
+
+        public static IHostBuilder CreateHostBuilder(string[] args) =>
+            Host.CreateDefaultBuilder(args)
+                .ConfigureWebHostDefaults(webBuilder =>
+                {
+                    webBuilder.UseStartup<Startup>();
+                })
+                .ConfigureLogging(builder =>
+                {
+                    builder.AddJsonConsole(o =>
+                    {
+                        o.IncludeScopes = false;
+                        o.TimestampFormat = "hh:mm:ss ";
+                        o.JsonWriterOptions = new JsonWriterOptions()
+                        {
+                            Encoder = JavaScriptTestEncoder.UnsafeRelaxedJsonEscaping, // TODO: 2
+                            Indented = true
+                        };
+                    });
+                });
+    }
+}
+```
+
+When running the app again, with the above change, the log message would look something like the snippet shown below:
+
+```
+{
+  "Timestamp": "09:08:33 ",
+  "EventId": 0,
+  "LogLevel": "Information",
+  "Category": "Microsoft.Hosting.Lifetime",
+  "Message": "Now listening on: https://localhost:5001",
+  "State": {
+    "Message": "Now listening on: https://localhost:5001",
+    "address": "https://localhost:5001",
+    "{OriginalFormat}": "Now listening on: {address}"
+  }
+}
+```
+
+Json console formatter by default logs each message in a single line, in order to make it more readable while configuring the formatter, we set Indented property on JsonWriterOptions to true.
+
+### How to select which formatter to use (via both code and config)?
+
+So far the samples showed how to register a formatter via code, but another way this can be done is via configuration. Looking back at the ASP.NET Core application sample, if we update the `appsettings.json` file rather than adding ConfigureLogging in Program.cs we could get the same outcome. Here's what we would add in the `appsettings.json` file:
```

Comment:
```suggestion
So far the samples showed how to register a formatter via code, but another way this can be done is via configuration. Looking back at the ASP.NET Core application sample, if we update the *appsettings.json* file rather than adding ConfigureLogging in *Program.cs*, we could get the same outcome. Here's what we would add in the *appsettings.json* file:
```